src/library/library.cpp

```diff
@@ -628,6 +644,32 @@ void Library::saveEvictedTrack(Track* pTrack) noexcept {
     }
 }
 
+void Library::exportTrackMetadata(
+        Track* pTrack,
+        TrackMetadataExportMode mode) const {
+    DEBUG_ASSERT(pTrack);
+
+    // Write audio meta data, if explicitly requested by the user
+    // for individual tracks or enabled in the preferences for all
+    // tracks.
+    //
+    // This must be done before updating the database, because
+    // a timestamp is used to keep track of when metadata has been
+    // last synchronized. Exporting metadata will update this time
+    // stamp on the track object!
+    if (pTrack->isMarkedForMetadataExport() ||
+            (pTrack->isDirty() && m_pConfig && m_pConfig->getValueString(ConfigKey("[Library]","SyncTrackMetadataExport")).toInt() == 1)) {
+        switch (mode) {
+        case TrackMetadataExportMode::Immediate:
+            SoundSourceProxy::exportTrackMetadataBeforeSaving(pTrack);
+            break;
+        case TrackMetadataExportMode::Deferred:
+            pTrack->markForMetadataExport();
```

Comment:
Yes. If the metadata is saved intermediately for keeping the databases up-to-date we must remember to save the file tags when the track objects goes out of scope. This is required even if the dirty flag is not set again, because no further changes happened after modifications have been saved to the database.

Reply:
I will add a comment.